sklearn/ensemble/_hist_gradient_boosting/tests/test_loss.py

```diff
@@ -68,7 +70,13 @@ def test_derivatives(loss, x0, y_true):
     get_gradients, get_hessians = get_derivatives_helper(loss)
 
     def func(x):
-        return loss.pointwise_loss(y_true, x)
+        if type(loss) == _LOSSES['binary_crossentropy']:
+            # Subtract a constant term such that the binary cross entropy
+            # has its minimum at zero. This only works if 0 < y_true < 1.
+            return loss.pointwise_loss(y_true, x) \
+                - loss.pointwise_loss(y_true, logit(y_true))
```

Comment:
I feel this is a bit complex.

Can we rather write, `
```py
if isinstance(loss, _LOSSES['binary_crossentropy']):
    assert_allclose(loss.pointwise_loss(y_true, optimum), 
                    loss.pointwise_loss(y_true, logit(y_true)))
else:
    ...
```
in the test directly?

Reply:
Good idea. Unfortunately, the `newton` root finder is allergic to such chances and complains with a no converged message as it can't ever reach a true zero. I would have to change from `newton` to a minimizer. I thought my change is the smaller one and the advantage of `newton` is that you also check for gradient and hessian in some way being correct. 
What do you think?